crates/kilt-dip-primitives/src/merkle_proofs/v0/provider_state/mod.rs

```diff
@@ -466,10 +468,11 @@ impl<
 			self.dip_proof.blinded.as_slice(),
 			proof_leaves_key_value_pairs.as_slice(),
 		)
+		// Can't log since the result returned by `verify_trie_proof` implements `Debug` only with `std`.
 		.map_err(|_| Error::InvalidDidMerkleProof)?;
 
 		let revealed_leaves = BoundedVec::try_from(self.dip_proof.revealed).map_err(|_| {
-			log::error!("Should not fail to construct BoundedVec since bounds were checked before.");
+			log::error!(target: "dip::consumer::DipDidProofWithVerifiedSubjectCommitmentV0", "Failed to construct BoundedVec<u8, {MAX_REVEALED_LEAVES_COUNT}>.");
```

Comment:
Do you think it could help to log, the actual size of the dip_proof.revealed length and the max size? Are there more possibilities, how this can fail?

Reply:
I mean, for those we should hopefully have access to the tx payload itself, which reveals what the input was. I tried to log only intermediate computations that are not directly visible by looking at a transaction input or output, mostly.

Reply:
In this case, `revealed` is provided by the user in the proof, so we know beforehand what its length is.